src/renderer/screens/exchange/Swap2/utils/shared/hooks.js

```diff
@@ -134,3 +134,35 @@ export const usePickExchangeRate = ({ exchangeRates, exchangeRate, setExchangeRa
     // eslint-disable-next-line
   }, [exchangeRates]);
 };
+
+// Poll the server to update the KYC status of a given provider.
+const KYC_STATUS_POLLING_INTERVAL = 10000;
+type kycItem = {
+  id: string,
+  status: string,
+};
+export const usePollKYCStatus = (
+  { provider, kyc, onChange }: { provider: string, kyc: kycItem, onChange: kycItem => void },
+  dependencies: any[] = [],
+) => {
+  useEffect(
+    () => {
+      if (kyc?.status !== "pending") return;
```

Comment:
Can you sync with @krauscher-ledger whether the status are expected to be stable if we add more providers, and if so, can we extract them into an enum instead of strings? That'd be pretty great to avoid me putting "peding" instead of "pending" which is only a matter of time

Reply:
I think you can assume it will be. We already "aggregate" the provider statuses in an enum on our side.
https://github.com/LedgerHQ/swap/blob/master/model/src/main/scala/co/ledger/swap/model/SwapStatus.scala

If we ever need to add more statuses, I'm fine with us handling it as a breaking change backend side (so new api version and maintained backward compatibility for 6 months).